### PR TITLE
[#884] Add prettier

### LINKED ISSUE #884: Add Prettier
### Context

[Prettier](https://github.com/prettier/prettier) auto-formats code before commit (or on save) to code standards set for the project. This should make code style a non-issue for new contributors and for the team.

### Proposal

Implement Prettier with our current TSLint rules.

1. Add the git hook using husky, as outlined in their README
2. Recommend using Prettier VSCode plug-in (optional) in our README
3. Remove lint step from grunt

### PR DESCRIPTION
**Pull request by LoserAntbear**
Closes #884 

## Changes:

- Prettier package added and configured, using extension of `./tslint/tslint.json`.
- Husky package added and `pre-commit` hook configured.
- `jasmine.d.ts` type definitions are now for `Jasmine 2.8.0.` It equals to the version used in current development.
- Added jasmine-core package from `v2.8.0` accordingly to version it should use.

### CODE REVIEW

**Comment by LoserAntbear** [comment]
I hope i did this right.

File: GruntFile.js [hunk 1]
```diff
@@ -340,7 +340,7 @@ module.exports = function (grunt) {
    //
 
    // Default task - compile & test
-   grunt.registerTask('default', ['tslint:src', 'compile', 'tests', 'visual']);
+   grunt.registerTask('default', ['compile', 'tests', 'visual']);
```
Comment (kamranayub) [inline GruntFile.js]: Wouldn't we still want this? We still want to lint before compiling. Prettier will only do formatting but there could still be other errors.

Comment (LoserAntbear) [reply GruntFile.js]: Sorry, suppose i got you wrong.
But which task you meant to remove there:  "3. Remove lint step from grunt"? 
This one was only linter task i found in grunt conf.

Comment (LoserAntbear) [reply GruntFile.js]: Yep. After surfing for i while, i found this: https://github.com/prettier/prettier/issues/2447

Looks like configs are unextendable and they are not going to add this feature.  So, i just configured prettier the same as linter.

**Comment by jedeen** [comment]
Thanks for this contribution! We'll merge this towards the end of this milestone, in order to standardize our formatting and prevent conflicts in future formatted files.

**Comment by kamranayub** [comment]
Yeah I'd say when milestone is ready before release we run this on the
whole codebase to prep it.

**Comment by LoserAntbear** [comment]
Awesome! Then, then, i'll remove same changes from #971 . And'll continue working on.

**Comment by eonarheim** [comment]
Thanks again @LoserAntbear!

**Pull request merged**